AK/LexicalPath.h

```diff
@@ -29,9 +29,21 @@ class LexicalPath {
 
     bool has_extension(const StringView&) const;
 
+    void append(String const& component);
+
     static String canonicalized_path(String);
     static String relative_path(String absolute_path, String const& prefix);
 
+    template<typename... S>
+    static LexicalPath join(String const& first, S... rest)
+    {
+        StringBuilder builder;
+        builder.append(first);
+        ((builder.append('/'), builder.append(rest)), ...);
+
+        return LexicalPath { builder.to_string() };
+    }
```

Comment:
I'd recommend making this API take a forwarding reference to `rest`:

```suggestion
    template<typename... S>
    static LexicalPath join(String const& first, S&&... rest)
    {
        StringBuilder builder;
        builder.append(first);
        ((builder.append('/'), builder.append(forward<S>(rest))), ...);

        return LexicalPath { builder.to_string() };
    }
```

Otherwise, the following block:
```c++
String a { "abc" };
String b { "def" };
auto path = LexicalPath::join(a, b);
```

Creates an extra copy of `b` when `join` is invoked. I think copies of `String` are cheapish (they just copy a `RefPtr`, right?) but perfectly-forwarding APIs is a good habit to get into anyways.

Reply:
Thanks for the suggestion!